### PR TITLE
Istanbul throws "Transformation error" exceptions - Closes #2858

### LINKED ISSUE #2858: Istanbul throws "Transformation error" exceptions
### Expected behavior

We should use nyc since istanbul is not being maintained anymore.

### Actual behavior

We are currently using istanbul, which causes the exceptions below:

image

### PR DESCRIPTION
**Pull request by yatki**
### What was the problem?
We were using outdated `istanbul` client which was causing babel transformation exceptions
### How did I fix it?

I replaced istanbul with nyc, updated `parallel_tests.js` file to execute tests with nyc, updated .gitignore and mocha.opts files to support nyc cli options.

### Review checklist

* The PR resolves #2858 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -103,14 +103,14 @@
 		"grunt-exec": "3.0.0",
 		"http-server": "0.11.1",
 		"husky": "1.3.1",
-		"istanbul": "0.4.5",
 		"istanbul-middleware": "0.2.2",
```
Comment (shuse2) [inline package.json]: should we don something about this `instanbul-middleware` too?

File: test/mocha.opts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 --timeout 1200s
 --exit
 --colors
---require ./test/setup.js
+--require test/setup.js
```
Comment (shuse2) [inline test/mocha.opts]: why did you change here...?

File: Jenkinsfile [hunk 3]
```diff
@@ -54,10 +54,10 @@ def teardown(test_name) {
 		nvm(getNodejsVersion()) {
 			sh """
 			rm -rf coverage_${test_name}; mkdir -p coverage_${test_name}
-			./node_modules/.bin/istanbul report --root ./test/.coverage-unit/ --dir ./test/.coverage-unit/
-			cp test/.coverage-unit/lcov.info coverage_${test_name}/ || true
-			./node_modules/.bin/istanbul report cobertura --root ./test/.coverage-unit/ --dir ./test/.coverage-unit/
-			cp test/.coverage-unit/cobertura-coverage.xml coverage_${test_name}/ || true
+			./node_modules/.bin/nyc report --report-dir framework/test/mocha/.nyc_output/
+			cp framework/test/mocha/.nyc_output/lcov.info coverage_${test_name}/ || true
```
Comment (fchavant) [inline Jenkinsfile]: This does not seem to work:
```
+ cp framework/test/mocha/.nyc_output/lcov.info coverage_get/
cp: cannot stat 'framework/test/mocha/.nyc_output/lcov.info': No such file or directory
[...]
+ cp framework/test/mocha/.nyc_output/lcov.info coverage_ws/
cp: cannot stat 'framework/test/mocha/.nyc_output/lcov.info': No such file or directory
[...]
+ cp framework/test/mocha/.nyc_output/lcov.info coverage_post/
cp: cannot stat 'framework/test/mocha/.nyc_output/lcov.info': No such file or directory
[...]
+ cp framework/test/mocha/.nyc_output/lcov.info coverage_integration/
cp: cannot stat 'framework/test/mocha/.nyc_output/lcov.info': No such file or directory
[...]
+ cp framework/test/mocha/.nyc_output/lcov.info coverage_unit/
cp: cannot stat 'framework/test/mocha/.nyc_output/lcov.info': No such file or directory
```

File: package.json (hunk 1 above)
Comment (yatki) [reply package.json]: I wanted to get rid of it too, but seems like it is being used for functional test coverage. We have to touch how we instantiate the application to remove istanbul-middleware, which I think should be done in a separate issue.

Or maybe we won't have to touch it once we write the new tests for new structure.

File: test/mocha.opts (hunk 2 above)
Comment (yatki) [reply test/mocha.opts]: To make paths uniform. I didn't use `./` for other paths too.

File: package.json (hunk 1 above)
Comment (yatki) [reply package.json]: @shuse2 created https://github.com/LiskHQ/lisk/issues/2919

File: Jenkinsfile [hunk 4]
```diff
@@ -54,12 +54,11 @@ def teardown(test_name) {
...
 			rm -rf coverage_${test_name}; mkdir -p coverage_${test_name}
-			./node_modules/.bin/istanbul report --root ./test/.coverage-unit/ --dir ./test/.coverage-unit/
-			cp test/.coverage-unit/lcov.info coverage_${test_name}/ || true
-			./node_modules/.bin/istanbul report cobertura --root ./test/.coverage-unit/ --dir ./test/.coverage-unit/
-			cp test/.coverage-unit/cobertura-coverage.xml coverage_${test_name}/ || true
+			./node_modules/.bin/nyc report --report-dir coverage_${test_name} --temp-dir framework/test/mocha/.nyc_output/${test_name}
+			./node_modules/.bin/nyc report --reporter=cobertura --report-dir coverage_${test_name} --temp-dir framework/test/mocha/.nyc_output/${test_name}
 			curl --silent http://localhost:4000/coverage/download --output functional-coverage.zip
 			unzip functional-coverage.zip lcov.info -d coverage_${test_name}/functional/
+			rm -rf framework/test/mocha/.nyc_output/${test_name}
```
Comment (fchavant) [inline Jenkinsfile]: Maybe we could move this 2 lines up with the other `nyc` related commands.

**Comment by yatki** [comment]
> @yatki I am still seeing `Transformation error`s in the the log file (for instance https://jenkins.lisk.io/job/lisk-core/job/PR-2869/48/artifact/lisk_functional_get.stdout.txt); is this normal?

@fchavant I think this is coming from `istanbul-middleware`. We are still using it to get functional test coverage. I created https://github.com/LiskHQ/lisk/issues/2919 for that.

**Comment by fchavant** [comment]
Another thing:
> [Cobertura] No coverage results were found using the pattern 'coverage_*/cobertura-coverage.xml' relative to '/home/lisk/workspace/lisk-core_PR-2869'.  Did you enter a pattern relative to the correct directory?  Did you generate the XML report(s) for Cobertura?

Maybe the file name changed with `nyc`?

**Comment by yatki** [comment]
We are going to put this PR on hold, because it took a lot of time and requires more research and thought.
